### PR TITLE
[browser] Allow leaving private mode with no tabs. Fixes JB#29800

### PR DESCRIPTION
**Pull request by jpetrell**

### CODE REVIEW

File: apps/browser/qml/pages/BrowserPage.qml [hunk 1]
```diff
@@ -244,8 +244,15 @@ Page {
 
         MouseArea {
             anchors.fill: parent
-            enabled: overlay.animator.atTop && webView.tabModel.count > 0
-            onClicked: overlay.dismiss(true)
+
+            property bool inEmptyPrivateMode: webView.privateMode && webView.privateTabModel.count === 0 && webView.persistentTabModel.count > 0
```
Comment (rainemak) [inline apps/browser/qml/pages/BrowserPage.qml:248]:
```suggestion
            readonly property bool inEmptyPrivateMode: webView.privateMode && webView.privateTabModel.count === 0 && webView.persistentTabModel.count > 0
```

Comment (rainemak) [reply apps/browser/qml/pages/BrowserPage.qml:248]: Move also anchors:fill: parent after property declaration.

Comment (jpetrell) [reply apps/browser/qml/pages/BrowserPage.qml:244]: Done

File: apps/browser/qml/pages/BrowserPage.qml [hunk 2]
```diff
@@ -244,8 +244,15 @@ Page {
...
+            enabled: overlay.animator.atTop && (webView.tabModel.count > 0 || inEmptyPrivateMode)
+            onClicked: {
+                if (inEmptyPrivateMode) {
```
[the remark this answers is not in the document]
Comment (jpetrell) [reply apps/browser/qml/pages/BrowserPage.qml:249]: True. Added Notice to inform user what is happening so leaving private mode does not come as surprise.

**Review by rainemak — APPROVED** [review]
LGTM, please rebase before merging.

**Comment by jpetrell** [comment]
Rebased.

**Pull request merged**